stdlib/public/SDK/Metal/Metal.swift

```diff
@@ -100,10 +100,11 @@ extension MTLDevice {
 @available(swift 4)
 @available(macOS 10.13, *)
 public func MTLCopyAllDevicesWithObserver(handler: @escaping MTLDeviceNotificationHandler) -> (devices:[MTLDevice], observer:NSObject) {
-    var resultTuple: (devices:[MTLDevice], observer:NSObject)
-    resultTuple.observer = NSObject()
-    resultTuple.devices = __MTLCopyAllDevicesWithObserver(AutoreleasingUnsafeMutablePointer<NSObjectProtocol?>(&resultTuple.observer), handler)
-    return resultTuple
+    var observer: NSObjectProtocol?
+    let devices = __MTLCopyAllDevicesWithObserver(&observer, handler)
+    // FIX-ME: The force cast here isn't great – ideally we would return the
+    // observer as an NSObjectProtocol.
+    return (devices, observer as! NSObject)
```

Comment:
As said in the comment, the force cast here isn't ideal, but it's better than what the code was previously doing – which was type punning an `NSObject` pointer to a `NSObjectProtocol?` pointer.

Reply:
Also, can I confirm that `__MTLCopyAllDevicesWithObserver` is guaranteed to set `observer`? The previous code was initially assigning `NSObject()`, but I'm not sure whether that was specifically because it could actually return that as the observer.

Reply:
Pinged Metal team about that.

Reply:
Here is the reply:

>`MTLCopyAllDevicesWithObserver(id <NSObject>* observer, MTLDeviceNotificationHandler handler)` allocates a framework object to return via the observer pointer (specifically a `_MTLDeviceNotifier` object), so the pointer passed in can be set to `nullptr`.